VimeoNetworking/Sources/NSError+Extensions.swift

```diff
@@ -85,18 +83,18 @@ public extension NSError
      
      - returns: a new `NSError`
      */
-    class func errorWithDomain(domain: String?, code: Int?, description: String?) -> NSError
+    class func errorWith(domain: String?, code: Int?, description: String?) -> NSError
     {
         var error = NSError(domain: VimeoErrorKey.VimeoErrorDomain.rawValue, code: 0, userInfo: nil)
         
         if let description = description
         {
             let userInfo = [NSLocalizedDescriptionKey: description]
-            error = error.errorByAddingDomain(domain, code: code, userInfo: userInfo)
+            error = error.errorByAdding(domain: domain, code: code, userInfo: userInfo)
```

Comment:
This is another weird one. This is an instance method on an NSError extension, which will always return a new instance of an NSError. But it reads real weird.

```Swift
// This…

func errorByAdding(domain: String?, code: Int?, userInfo: [AnyHashable: Any]?) -> NSError

// …reads like this at the call site:

let error = someError.errorByAdding(domain: someDomain, code: someCode, userInfo: someUserInfo)
```

Using your suggestion we get this:

```Swift 
func error(byAddingDomain domain: String?, code: Int?, userInfo: [AnyHashable: Any]?) -> NSError

// …which produces this at the call site:

let error = someError.error(domain: someDomain, code: someCode, userInfo: someUserInfo)
```

Which still has the problem of being kind of ambiguous. It's not clear that the point of the method is that we're _adding_ anything.

I'd almost rather do something like this:

```Swift 
func byAddingDomain(_ domain: String?, code: Int?, userInfo: [AnyHashable: Any]?) -> NSError

// …which give us this:

let error = someError.byAddingDomain(someDomain, code: someCode, userInfo: someUserInfo)
```